//===== Hercules Patch =======================================
//= 1charpergm
//===== By: ==================================================
//= AnnieRuru
//===== Current Version: =====================================
//= 0.2
//===== Compatible With: ===================================== 
//= Hercules 2019-02-02
//===== Description: =========================================
//= make each GM account can only have 1 character slot
//===== Topic ================================================
//= http://herc.ws/board/topic/11215-one-char-per-gm-account/?do=findComment&comment=90332
//===== Additional Comments: =================================  
//= simple trick, only allow them to create/login the character on slot 0
//============================================================

 src/char/char.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/char/char.c b/src/char/char.c
index 330b0639f..d61203407 100644
--- a/src/char/char.c
+++ b/src/char/char.c
@@ -1729,6 +1729,9 @@ static int char_make_new_char_sql(struct char_session_data *sd, const char *name
 			return -2; // Char Creation Denied
 	}
 
+	if ( slot != 0 && sd->group_id > 10 )
+		return -2;
+
 	//check other inputs
 #if PACKETVER >= 20120307
 	if(slot < 0 || slot >= sd->char_slots)
@@ -4565,6 +4568,11 @@ static void char_parse_char_select(int fd, struct char_session_data *sd, uint32
 
 	RFIFOSKIP(fd,3);
 
+	if ( slot != 0 && sd->group_id > 10 ) {
+		chr->creation_failed(fd, -4);
+		return;
+	}
+
 #if PACKETVER >= 20110309
 	if( pincode->enabled ){ // hack check
 		struct online_char_data* character;